File: android/Girlsfleet/app/src/main/java/com/circus/girlsfleet/FPSHelper.java
package com.circus.girlsfleet;

public class FPSHelper {

        public static boolean waitFrame(long prevFrameTime,long prevViewUpdateTime
                ,long currentTime){

            long timeElapsed = currentTime - prevFrameTime;
            if (Config.TIME_PER_FRAME > timeElapsed) {
                try {
                    Thread.sleep(Config.TIME_PER_FRAME - timeElapsed);
                } catch (Exception e) {
                    e.printStackTrace();
                }
                return true;
            }
            if (0L == timeElapsed)
                return true;
            if (Config.TIME_PER_FRAME * 4 <= (currentTime - prevViewUpdateTime)) {
                return true;
            }

            return false;

        }
}

File: android/Circus/app/src/main/java/com/personal/circus/FPSHelper.kt
package com.personal.circus;

import java.lang.Exception

class FPSHelper {

    companion object {
        @JvmStatic
        fun waitFrame(prevFrameTime: Long, prevViewUpdateTime: Long, currentTime: Long): Boolean {
            val timeElapsed = currentTime - prevFrameTime;
            if (Config.TIME_PER_FRAME > timeElapsed) {
                try {
                    Thread.sleep(Config.TIME_PER_FRAME - timeElapsed);
                } catch (e: Exception) {
                }
                return true;
            }
            if (0.toLong() == timeElapsed)
                return true;
            if (Config.TIME_PER_FRAME * 4 <= (currentTime - prevViewUpdateTime)) {
                return true;
            }

            return false;

        }

    }
}